conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/EnclaveEnvironment.kt

```diff
@@ -129,6 +132,46 @@ abstract class EnclaveEnvironment(enclaveProperties: Properties, kdsConfig: Encl
         inMemoryMountPath: String,
         persistentMountPath: String,
         encryptionKey: ByteArray)
+
+    /** Call interface functions */
+    /**
+     * Send enclave info to the host.
+     * TODO: It would be better to return enclave info from the initialise enclave call
+     *       but that doesn't work in mock mode at the moment.
+     */
+    fun setEnclaveInfo(signatureKey: PublicKey, encryptionKeyPair: KeyPair) {
+        val encodedSigningKey = signatureKey.encoded                    // 44 bytes
+        val encodedEncryptionKey = encryptionKeyPair.public.encoded     // 32 bytes
+        val payloadSize = encodedSigningKey.size + encodedEncryptionKey.size
+        val buffer = ByteBuffer.allocate(payloadSize).apply {
+            put(encodedSigningKey)
+            put(encodedEncryptionKey)
+        }
+        hostInterface.executeOutgoingCall(HostCallType.SET_ENCLAVE_INFO, buffer)
+    }
+
+    /**
+     * Get a signed quote from the host.
+     */
+    fun getSignedQuote(report: ByteCursor<SgxReport>): ByteCursor<SgxSignedQuote> {
```

Comment:
This function is only used in Native mode I believe. Do we need it here? It is getting inherited